pkg/services/search/service.go

```diff
@@ -58,10 +58,24 @@ type SearchService struct {
 }
 
 func (s *SearchService) SearchHandler(ctx context.Context, query *Query) error {
+	starredQuery := star.GetUserStarsQuery{
+		UserID: query.SignedInUser.UserID,
+	}
+	staredDashIDs, err := s.starService.GetByUser(ctx, &starredQuery)
+	if err != nil {
+		return err
+	}
+
+	// filter by starred dashboard IDs when starred dashboards are requested and no UID or ID filters are specified to improve query performance
+	if query.IsStarred && len(query.DashboardIds) == 0 && len(query.DashboardUIDs) == 0 {
+		for id := range staredDashIDs.UserStars {
```

Comment:
Nit: We can short circuit the request here if we don't have any starred dashboards

Reply:
Good shout, added the change.